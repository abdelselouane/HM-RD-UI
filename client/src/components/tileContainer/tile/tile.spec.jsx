/* global describe, beforeAll, jest, it, expect */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Tile from './tile';

describe('<Tile />', () => {
  let wrapper;
  let spy;
  let tile;

  beforeAll(() => {
    tile = {
      title: 'Mock Tile',
      className: 'mock-class',
      icon: 'mock-icon',
      index: 0,
      handleMouseEnter: jest.fn(),
      handleTileFunction: jest.fn(),
    };

    wrapper = shallow(<Tile {...tile} />);
  });

  it('should render the tableRow', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call handleMouseEnter method on mouse enter of tile', () => {
    spy = jest.spyOn(tile, 'handleMouseEnter');
    wrapper.find('.mock-class').simulate('mouseEnter');
    expect(spy).toHaveBeenCalled();
    spy.mockReset();
    spy.mockRestore();
  });

  it('should call handleTileFunction method on click of tile', () => {
    spy = jest.spyOn(tile, 'handleTileFunction');
    wrapper.find('.mock-class').simulate('click');
    expect(spy).toHaveBeenCalled();
    spy.mockReset();
    spy.mockRestore();
  });
});
