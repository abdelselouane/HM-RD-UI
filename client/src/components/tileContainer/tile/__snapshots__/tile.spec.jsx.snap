// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tile /> should render the tableRow 1`] = `
<div
  className="card repair-tile mock-class"
  onClick={[MockFunction]}
  onMouseEnter={[Function]}
  tabIndex={0}
>
  <div
    className="card-toolbar"
  >
    <span
      className="card-title"
    >
      <h2>
        Mock Tile
      </h2>
    </span>
  </div>
  <div
    className="card-content"
  >
    <i
      className="icon_mock-icon"
    />
  </div>
</div>
`;
